mysql/content.md

```diff
@@ -152,6 +152,24 @@ If there is no database initialized when the container starts, then a default da
 
 If the application you're trying to connect to MySQL does not handle MySQL downtime or waiting for MySQL to start gracefully, then a putting a connect-retry loop before the service starts might be necessary. For an example of such an implementation in the official images, see [WordPress](https://github.com/docker-library/wordpress/blob/1b48b4bccd7adb0f7ea1431c7b470a40e186f3da/docker-entrypoint.sh#L195-L235) or [Bonita](https://github.com/docker-library/docs/blob/9660a0cccb87d8db842f33bc0578d769caaf3ba9/bonita/stack.yml#L28-L44).
 
+If you are using `docker-compose`, then you may also use a healthcheck on the MySQL container to ensure it is accepting connections before starting other containers which depend on it:
+
+```
+services:
+  mysql:
+    image: mysql:8.0.15
+    environment:
+      MYSQL_ROOT_PASSWORD: example
+    healthcheck:
+      test: 'mysql -e SELECT\ 1'
+                                                                                 
+  app:
+    ...
+    depends_on:
+      mysql:
+        condition: service_healthy
```

Comment:
Given the caveats at https://docs.docker.com/compose/compose-file/#depends_on, especially the following one, I'm wary of adding anything that references `depends_on:` in the documentation:

> - Version 3 no longer supports the `condition` form of `depends_on`.
> 
> - The `depends_on` option is ignored when deploying a stack in swarm mode with a version 3 Compose file.

Reply:
That's a good point, this only works with compose file formats 2.1 - 2.4. I think it is worth sharing, but this probably isn't the best way to do that.